### PR TITLE
10.0 ms sync conflicts imp

### PR DESCRIPTION
**Pull request by GabbasovDinar**

### CODE REVIEW

File: pos_multi_session/static/src/js/pos_multi_session.js [hunk 1]
```diff
@@ -140,10 +140,9 @@ odoo.define('pos_multi_session', function(require){
                     order.order_removing_to_send();
                 });
                 var channel_name = "pos.multi_session";
-                var callback = self.updates_from_server;
+                var callback = self.ms_update_callback;
```
Comment (yelizariev) [inline pos_multi_session/static/src/js/pos_multi_session.js]: updates_from_server_callback?

File: pos_multi_session/static/src/js/pos_multi_session.js [hunk 2]
```diff
@@ -226,8 +225,25 @@ odoo.define('pos_multi_session', function(require){
...
+            }
+            return this.updates_from_server(message, sync_all);
+        },
         updates_from_server: function(message, sync_all){
             // don't broadcast updates made from this message
+            if (message.login_number === this.pos_session.login_number) {
+                // we don't process updates were send from this device
+                // keep the same message_ID among the same POS to prevent endless sync_all requests
+                this.message_ID = message.data.message_ID;
+                return;
```
Comment (yelizariev) [inline pos_multi_session/static/src/js/pos_multi_session.js]: :+1:

**Review by yelizariev — COMMENTED** [review]
Rename

File: pos_multi_session/static/src/js/pos_multi_session.js (hunk 1 above)
Comment (KolushovAlexandr) [reply pos_multi_session/static/src/js/pos_multi_session.js]: yep, it were done before the renamings

File: pos_multi_session/static/src/js/pos_multi_session.js [hunk 3]
```diff
@@ -171,6 +171,15 @@ odoo.define('pos_multi_session', function(require){
 
                 return self.multi_session.request_sync_all({'immediate_rerendering': true}).then(function() {
                     done.resolve();
+                }).fail(function(){
+                    var request_interval = setInterval(function(){
+                        return self.multi_session.request_sync_all({'immediate_rerendering': true}).then(function() {
+                            done.resolve();
+                        });
+                    }, 2000);
```
Comment (yelizariev) [inline pos_multi_session/static/src/js/pos_multi_session.js]: Naive approach. Make more stable solution

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Pull request merged**